docs/source/usage_guides/deepspeed_multiple_model.md

```diff
@@ -0,0 +1,254 @@
+
+```python
+active_plugin = get_active_deepspeed_plugin(accelerator.state)
+assert active_plugin is deepspeed_plugins["student"]
+```
+
+Also the `AcceleratorState` will keep the active DeepSpeed plugin saved in `state.deepspeed_plugin`:
+```python
+assert active_plugin is accelerator.deepspeed_plugin
+```
+
+Since the `student` is the currently active plugin, let's go ahead and prepare our model, optimizer, and scheduler like normal:
+
+```python
+student_model, optimizer, scheduler = ...
+student_model, optimizer, scheduler, train_dataloader = accelerator.prepare(student_model, optimizer, scheduler, train_dataloader)
+```
+
+From here, we need to deal with the teacher model.
+
+### The teacher model
+
+First we need to tell the `Accelerator` that the zero3/teacher ZeRO configuration should be used:
+
+```python
+accelerator.state.select_deepspeed_plugin("teacher")
+```
+
+Doing so will disable the `"student"` plugin and enable the `"teacher"` one instead. This will update the
+DeepSpeed stateful config inside of `transformers`, and change which plugin configuration gets called when using
+`deepspeed.initialize()`, allowing us to use the no-code `deepspeed.zero.Init`  context manager `transformers` provides:
+
+```python
+teacher_model = AutoModel.from_pretrained(...)
+teacher_model = accelerator.prepare(teacher_model)
+```
+
+Otherwise you should manually initialize them under `deepspeed.zero.Init`:
+```python
+with deepspeed.zero.Init(accelerator.deepspeed_plugin.config):
+    model = MyModel(...)
+```
+
+### Training
+
+From here, your training loop can be whatever you like, making sure that `teacher_model` is never being trained on:
+
+```python
+teacher_model.eval()
+student_model.train()
+for batch in train_dataloader:
+    with torch.no_grad():
+        output_teacher = teacher_model(**batch)
+    output_student = student_model(**batch)
+    # Combine the losses or modify it in some way
+    loss = output_teacher.loss + output_student.loss
+    accelerator.backward(loss)
+    optimizer.step()
+    scheduler.step()
+    optimizer.zero_grad()
+```
+
+## Multiple models all being trained
+
+Multiple models being trained is a more complicated scenario and is actively being looked upon by the team.
+In its current state, we assume that each model is **completely disjoint** from the other during training.
+
+With this situation, we still require two [`utils.DeepSpeedPlugin`]'s to be made, however we also further
+require a second `Accelerator`, since we need to call different `deepspeed` engines at different times, and a single `Accelerator` only
+carries one instance of it at a time.
+
+Since the [`state.AcceleratorState`] is a stateful object though, it is already aware of both `DeepSpeedPlugin`'s available,
+meaning you can just instantiate a second `Accelerator` with no extra arguments:
```

Comment:
```suggestion
Training multiple models is a more complicated scenario.
In its current state, we assume each model is **completely disjointed** from the other during training.

This scenario still requires two [`utils.DeepSpeedPlugin`]'s to be made. However, you also need a second [`Accelerator`], since different `deepspeed` engines are being called at different times. A single [`Accelerator`] can only carry one instance at a time.

Since the [`state.AcceleratorState`] is a stateful object though, it is already aware of both [`utils.DeepSpeedPlugin`]'s available. You can just instantiate a second [`Accelerator`] with no extra arguments.
```